### DISCUSSION TITLE
ExpressJS server structure

### DISCUSSION THREAD

**Discussion by gbowne1** [gbowne1/RadioLogger#34] [Ideas]
We can discuss ExpressJS server structure here

**Comment by jzunigarce** [comment]
The user schema only have 4 field(_id, name, email, passwor)?

**Reply by gbowne1** [reply]
I was not certain what actually goes here.  Other than those fields.  We should really only need a user name and password to login/signin.  Email should only be provided at registration time.  Although user could have plenty of fields.

**Reply by jzunigarce** [reply]
I propose that we start with these fields, since we are working with mongodb we are not anchored to a schema. Could we add these fields on the registration page?

**Reply by gbowne1** [reply]
yeah.  Fine with me. I think that would make the most sense initially. We can decide on what to do as we move forwards.

**Reply by gbowne1** [reply]
I can dropbox the contents of the old project if you like if there is anything you would like to see.

**Reply by jzunigarce** [reply]
Another question, will the requests from the client to the server (send and receive data) be asynchronous?

**Reply by gbowne1** [reply]
yes.  I couldn't think of any other method that might work.  We will just have to test and see.

**Comment by gbowne1** [comment]
I did not have much luck with the login page with any of the functions.  It didnt seem to redirect to dashboard once auth etc. Console wasnt much help either.

**Reply by jzunigarce** [reply]
On the weekend in work with loggin backend. I can help with front

**Reply by gbowne1** [reply]
Ok.  Let me know if you need anything.  The css never got finished for the login.

I didnt have much time today to work on this.

If we can make the dashboard's navbar reusable that will help. Then can work on SideNav drawer.

**Reply by jzunigarce** [reply]
Are all link fix in all pages?

**Reply by gbowne1** [reply]
They should be.  I changed most of the href in the dashboard to the correct page links.

**Reply by gbowne1** [reply]
I need to add a model for the QSL card db page...

const mongoose = require("mongoose");
const { Schema } = mongoose;

const QslCardSchema = new Schema({
  callsign: {
    type: String,
    required: true
  },
  contactDate: {
    type: Date,
    required: true
  },
  contactTime: {
    type: String,
    required: true
  },
  frequency: {
    type: Number,
    required: true
  },
  mode: {
    type: String,
    required: true
  },
  signalReport: {
    type: String,
    required: true
  },
  comments: {
    type: String,
    required: false
  }
});

module.exports = mongoose.model("QslCard", QslCardSchema);

**Reply by jzunigarce** [reply]
> They should be.  I changed most of the href in the dashboard to the correct page links.

But the other pages have error with link

**Reply by gbowne1** [reply]
Which ones? I left a couple of them not finished.  The lsst 2 cards I sm not sure yet whst to use them for.

**Reply by jzunigarce** [reply]
<img width="1234" alt="image" src="https://user-images.githubusercontent.com/722001/235256644-1df21ae6-335c-4e18-8653-9b7f3d28402e.png">

**Reply by gbowne1** [reply]
yeah, that's because there is a manifest.  that's needed for good SEO.  I dont know about the cors.  probably a bad SHA on this particular page.  (At some point we should be using downloaded assets from bootstrap, popperjs, fontawesome/fortawesome and hopefully sooner than later.) Usually its better to CDN for dev and download / build assets for prod.. unless the CDN is quick.  cloudflare is usually responsive.  

There are also no icons with those filenames.  Need to modify the names. I put the icons in the icons folder in the client.
